src/Type/Constant/ConstantArrayType.php

```diff
@@ -220,6 +220,32 @@ public function getValueTypes(): array
 		return $this->valueTypes;
 	}
 
+	public function getFirstValueType(): Type
```

Comment:
there's a `getValueTypes` right above which simply returns all valueTypes while this one is "smart" and returns the type that you would expect when you access the first or last element in PHP => it deals with optional keys correctly.

I was just afraid that naming-wise one might expect that this should simply return `$this->valueTypes[0]`. is the name OK, or does maybe `getFirstValue` or so make more sense? hmm

Reply:
in case the method is not re-usable/usefull as is for other use-cases you might move them as a private method into the extensions instead

Reply:
also possible, I agree, I'm fine with anything here I guess